### DISCUSSION TITLE
Source generators: How to distinguish between design-time build vs. full build

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/roslyn#57749] [Q&A] [answered]
The idea is to only generate method signatures in VS builds to provide autocomplete without unnecessary work.
```cs
public void GeneratedMethod() { throw new NotImplemented(); }
```
If a full build is requested we go ahead with generating the body.
```cs
public void GeneratedMethod() { /* actual impl */ }
```

**Comment by PathogenDavid** [comment, marked as answer]
I don't think there's a reliable way to do this today, but https://github.com/dotnet/roslyn/issues/57589 tracks adding an API for doing this.
